shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -224,48 +286,43 @@ class JSBackend {
             throw new Exception(s"Cannot have two base classes: $cls1, $cls2")
           }
       }
-    case _ =>
-      ???
+    // `class C: F[X]` and (`F[X]` => `A`) ==> `class C extends A {}`
+    // For applied types such as `Id[T]`, evaluate them before translation.
+    // Do not forget to evaluate type arguments first.
+    case AppliedType(Primitive(base), targs) =>
+      getBaseClassAndFields(applyTypeAlias(base, targs map { substitute(_) }))
+    // There is some other possibilities such as `class Fun[A]: A -> A`.
+    // But it is not achievable in JavaScript.
+    case _ => ???
   }
 
-  def translateClassDeclaration(name: Str, actualType: Type): JSClassDecl =
-    actualType match {
-      // `class A` ==> `class A {}`
-      case Top => JSClassDecl(name, Nil)
-      // `class A { <items> }` ==> `class A { constructor(fields) { <items> } }`
-      case Record(fields) => JSClassDecl(name, fields map { _._1 })
-      // `class B: A` ==> `class B extends A {}`
-      case Primitive(baseName) =>
-        resolvedAliases get baseName match {
-          case S(resolvedBaseName) =>
-            nameClsMap get resolvedBaseName match {
-              case N =>
-                throw new Error(s"Class $resolvedBaseName is not defined.")
-              case S(cls) => JSClassDecl(name, Nil, S(cls))
-            }
-          case N => throw new Exception(s"unresolved alias: $baseName")
-        }
-      // `class C: A & { <items> }` ==>
-      // `class C extends A { constructor(fields) { <items> } }`
-      case ty: Inter =>
-        val (fields, clsNmeOpt) = expandRecordInter(ty)
-        clsNmeOpt match {
-          case N => JSClassDecl(name, fields.distinct, N)
-          case S(clsNme) =>
-            nameClsMap get clsNme match {
-              case N      => throw new Error(s"Class $clsNme is not defined.")
-              case S(cls) => JSClassDecl(name, fields.distinct, S(cls))
-            }
+  // Translate MLscript class declaration to JavaScript class declaration.
+  // First, we will analyze its fields and base class name.
+  // Then, we will check if the base class exists.
+  def translateClassDeclaration(name: Str, actualType: Type): JSClassDecl = {
+    getBaseClassAndFields(actualType) match {
+      // Case 1: no base class, just fields.
+      case fields -> N => JSClassDecl(name, fields.distinct, N)
+      // Case 2: has a base class and fields.
+      case fields -> S(clsNme) =>
+        nameClsMap get clsNme match {
+          case N      => throw new Error(s"Class $clsNme is not defined.")
+          case S(cls) => JSClassDecl(name, fields.distinct, S(cls))
         }
-      // I noticed `class Fun[A]: A -> A` is okay.
-      // But it is not achievable in JavaScript.
-      case _ => ???
     }
+  }
 
   def apply(pgrm: Pgrm): Ls[Str] = {
-
     val (diags, (typeDefs, otherStmts)) = pgrm.desugared
-    resolveAliases(typeDefs)
+
+    // Collect type aliases into a map so we can evaluate them.
```

Comment:
I don't think `evaluate` is really the right terminology. Maybe just say "normalize"?
```suggestion
    // Collect type aliases into a map so we can normalize them.
```